File: python/4th_week/contohsoal1/Matriks.py
from numpy import mat


class Matriks:
    # implementasi class matriks
    # contruktor
    
    def __init__(self, baris, kolom):
        self.baris = baris
        self.kolom = kolom
        self.mat = [baris][kolom]

    # set and get baris
    def setBaris(self, paramBaris):
        self.baris = paramBaris
    
    def getBaris(self):
        return self.baris
    
    # set and get kolom
    def setKolom(self, paramKolom):
        self.kolom = paramKolom
    
    def getKolom(self):
        return self.kolom

    # set and get matriks
    def setMat(self):
        mat[self.baris][self.kolom]
    
    def getMat(self):
        return mat
    
    # set and get 
    def setSel(self, b, k, nilai):
        self.mat[b][k] = nilai
    
    def getSel(self, baris, kolom):
        return self.mat[baris][kolom]

File: cpp/4th_week/contohsoal1/Matriks.cpp
#include <iostream>
#include <string>

using namespace std;

class Matriks
{
private:
    /* data */
    int baris;
    int kolom;
    int mat[50][50];
public:
    // konstruktor
    Matriks() {}
    Matriks(int baris, int kolom) {
        this->baris = baris;
        this->kolom = kolom;
        mat[baris][kolom];
    }

    // set and get baris
    void setBaris(int baris) {
        this->baris = baris;
    }

    int getBaris() {
        return baris;
    }

    //set and get kolom
    void setKolom(int kolom) {
        this->kolom = kolom;
    }

    int getKolom() {
        return kolom;
    }

    void setMat(int baris, int kolom) {
        mat[baris][kolom];
    }

    int getMat() {
        return mat[baris][kolom];
    }

    //  set and get sel
    void setSel(int baris, int kolom, int nilai){
        mat[baris][kolom] = nilai;
    }

    int getSel(int baris, int kolom) {
        return mat[baris][kolom];
    }

    ~Matriks() {}
};